src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -964,7 +964,7 @@ private void WriteHeaders(HttpRequestMessage request)
             }
         }
 
-        private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request, CancellationToken cancellationToken, bool flush = false)
```

Comment:
Nit: flush => mustFlush or forceFlush or something like that, just to call out that it might still flush even if this is false.